Assets/Content/Items/Functional/Parts/Lights/BlinkingLight.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(MeshRenderer))]
+public class BlinkingLight : MonoBehaviour
+{
+    enum LightMode
+    {
+        StayOn,
+        Blinking,
+        Off
+    }
+    
+    private MeshRenderer lightMesh;
+    private LightMode lightMode = LightMode.Off;
+    private IEnumerator blinkCoroutine;
+    
+    [Header("Materials")]
+    [SerializeField] private Material nonEmissiveMaterial;
+    [SerializeField] private Material emissiveMaterial;
+
+    [Header("Blink Settings")]
+    [Tooltip("How often should a blink occur in seconds.")]
+    [SerializeField] private float blinkPeriod = 1f;
+    [Tooltip("How long should the light stay on when blinking. Must be less than blink period.")]
+    [SerializeField] private float blinkLength = .5f;
+    
+    void Awake() 
```

Comment:
Is there a reason why this is in `Awake` and not `Start`?

Reply:
I was testing it locally in another project and left that there. No reason. Switching it back to Start.